src/reconcile/rec_visibility.c

```diff
@@ -1150,7 +1320,9 @@ __wti_rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_INSERT *ins,
     /*
      * Paranoia: check that we didn't choose an update that has since been rolled back.
      */
-    WT_ASSERT_ALWAYS(session, upd_select->upd == NULL || upd_select->upd->txnid != WT_TXN_ABORTED,
+    WT_ASSERT_ALWAYS(session,
+      upd_select->upd == NULL || upd_select->upd->txnid != WT_TXN_ABORTED ||
+        F_ISSET(S2C(session), WT_CONN_PRESERVE_PREPARED),
```

Comment:
May be we can also again assert that the time window has prepare?

Reply:
done